bc_obps/reporting/migrations/0123_cleanup_orphaned_raw_activity_data.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 5.1.10 on 2025-07-31 10:40
+
+from django.db import migrations
+from common.lib import pgtrigger
+
+
+def cleanup_orphaned_raw_activity_data(apps, schema_editor):
+    """
+    Remove ReportRawActivityData records where the activity is not associated with the facility report.
+    """
+    ReportRawActivityData = apps.get_model('reporting', 'ReportRawActivityData')
+    FacilityReport = apps.get_model('reporting', 'FacilityReport')
+
+    # Get all raw activity data records
+    raw_data_records = ReportRawActivityData.objects.all()
+
+    records_to_delete = []
+
+    for raw_data in raw_data_records:
+        facility_report_id = raw_data.facility_report_id
+        activity_id = raw_data.activity_id
+
+        # Check if this facility report has this activity associated
+        facility_report = FacilityReport.objects.get(id=facility_report_id)
+        if not facility_report.activities.filter(id=activity_id).exists():
+            records_to_delete.append(raw_data.id)
+
+    # Delete orphaned records
+    if records_to_delete:
+        with pgtrigger.ignore("reporting.ReportRawActivityData:immutable_report_version"):
+            ReportRawActivityData.objects.filter(id__in=records_to_delete).delete()
+            print(f"Deleted {len(records_to_delete)} orphaned ReportRawActivityData records")
+    else:
+        print("No orphaned ReportRawActivityData records found")
+
+
+def reverse_cleanup_orphaned_raw_activity_data(apps, schema_editor):
+    """
+    This migration is irreversible as we cannot restore deleted orphaned data.
+    """
+    pass
```

Comment:
the preferred method for making a migration non-reversible is to use 

```
    operations = [
        migrations.RunPython(
            code=<function_name>,
            reverse_code=migrations.RunPython.noop,
            elidable=True,
        ),
    ]
```
   
see registration/migrations/0095 for an example. 
    
With the recommended approach, Django knows that the migration hasn't been reversed, whereas with the current script, it thinks it has applied the reversal (even though it did nothing). We should also add `elidable=True` in case we ever squash our migrations.